tests/features/java.security.feature

```diff
@@ -2,5 +2,5 @@
 @ubi9/openjdk-17
 Feature: Openshift S2I tests
   Scenario: Check networkaddress.cache.negative.ttl has been set correctly
-    Given s2i build https://github.com/jboss-openshift/openshift-examples/ from binary-cli-security-property
+    Given s2i build https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources from binary-cli-security-property
```

Comment:
This URI is not a valid Git repository. It results in the following error:
```
ERROR:cekit:E1024 13:55:37.768913  155305 git.go:188] Clone failed: source https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources#master, target /tmp/s2i648681403/upload/tmp, with output "fatal: repository 'https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources/' not found\n"
```

In order to make the behave tests reference a different git branch, you need to change the "Given" line, in a way like this:

```diff
-    Given s2i build https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources from undertow-servlet
+    Given s2i build https://github.com/jhuttana/openjdk-test-applications from undertow-servlet using pick_relevant_sources
```

The list of valid "Given" commands is controlled by the "behave-test-steps" repository. For example https://github.com/cekit/behave-test-steps/blob/v1/steps/s2i_steps.py#L53

For most of the "Given" command that we currently use, there should be an alternative that lets you specify the branch (confusingly phrased as "tag" in the behave-test-steps source.) There might be some tests where there is not an appropriate "GIven" command already: in which case, we can add one to behave-test-steps.

Reply:
Alternatively, if you configure your GitHub fork of openjdk-test-applications to make the branch "pick_relevant_sources" the default, you would not need to specify it in these tests.

Reply:
> This URI is not a valid Git repository. It results in the following error:
> 
> ```
> ERROR:cekit:E1024 13:55:37.768913  155305 git.go:188] Clone failed: source https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources#master, target /tmp/s2i648681403/upload/tmp, with output "fatal: repository 'https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources/' not found\n"
> ```
> 
> In order to make the behave tests reference a different git branch, you need to change the "Given" line, in a way like this:
> 
> ```diff
> -    Given s2i build https://github.com/jhuttana/openjdk-test-applications/tree/pick_relevant_sources from undertow-servlet
> +    Given s2i build https://github.com/jhuttana/openjdk-test-applications from undertow-servlet using pick_relevant_sources
> ```
> 
> The list of valid "Given" commands is controlled by the "behave-test-steps" repository. For example https://github.com/cekit/behave-test-steps/blob/v1/steps/s2i_steps.py#L53
> 
> For most of the "Given" command that we currently use, there should be an alternative that lets you specify the branch (confusingly phrased as "tag" in the behave-test-steps source.) There might be some tests where there is not an appropriate "GIven" command already: in which case, we can add one to behave-test-steps.

I have updated the URI in the "Given" section.

Reply:
> Alternatively, if you configure your GitHub fork of openjdk-test-applications to make the branch "pick_relevant_sources" the default, you would not need to specify it in these tests.

Thanks for letting me know about this :)